product_docs/docs/biganimal/release/using_cluster/managing_cloud_account/csp_tagging.mdx

```diff
@@ -0,0 +1,105 @@
+---
+title: "Tagging AWS resources"
+deepToC: true
+---
+
+Tags are key-value pairs you can apply to cloud provider resources created on your own cloud account (Your Cloud Account). Resource tagging allows you to have fine-grained access control and granular cost data at the bucket level. When you add a tag, the following resources are labeled: 
+
+- EC2 instances
+- S3 buckets
+- Load balancers
+- Volumes
+
+You can create and manage tags on a per-project basis in the EDB Postgres® AI Console. The Console then applies the tag to the resources of all regions configured in your account. 
+
+## Create Tags
+
+To create a tag:
+
+1.  Log in to the Console.
+    
+1.  Go to the project's page.
+
+1.  In the project's list, select your **Project**.
+
+1.  On your project's home page left navigation, select **Cloud Providers**.
+
+1.  On the cloud provider's page, select **Your Cloud Account**.
+
+1.  On your cloud account tab, under **AWS**, select **Manage Tags**.
+
+1.  Select **Add a Key Value pair**.
+
+1.  Provide **Key** and **Value** and select **Save**.
+
+1.  A **Confirm changes to AWS tags** message pops up.
+
+1.  Select **Confirm** to add the tag.
+
+1.  This **Key:Value** pair is added at your project level. It is populated on the EC2 instances, S3 buckets, load balancers and volumes managed in that project, for all regions and clusters.
+
+This new tag appears in the list under **AWS** on Cloud Service Providers. 
+
+This tag is also propagated on your AWS resources. For instance, you can view this tag in the **Tags** tab of your EC2 instance resources on the AWS console.
+
+### Considerations
```

Comment:
@amine-khemissi I reworked this section, let me know what you think. I'd prefer keeping the word considerations instead of limitations, as I don't expect many customers to encounter these issues, and limitations have a negative connotation, but I will bring this up with the docs team. 

@navnathedb, let me know if there are other BA reserved words we should add here.

Reply:
Along with reversed keys, this doc also needs to talk default tags applied on Cluster and Project resources

<img width="567" alt="Screenshot 2024-07-19 at 3 48 57 PM" src="https://github.com/user-attachments/assets/7fbe4d45-b2e7-4f59-b862-1bd3aa14c471">

ref: https://github.com/EnterpriseDB/upm-rfcs/blob/main/rfcs/rfc-upm-34349-csp-tag-controller.md#3-design

Reply:
Hi @dhope-nagesh  
Is it these four tags only? Or might there be other BA-generated tags? And do these tags appear in the BA and AWS dashboards as well? Or are they "hidden"? 

We already mention the existence of the BA default tags in the context of the maximum limitation being 50 tags per resource (line 53). So now I am wondering if we should provide a list of all default BA tags? But I think that depends on 1. whether these are visible to them or not and 2. if they need to know the names of the tags for other operations.

Reply:
These 4 tags are part of the requirement from the product team, but they are not visible on the BA portal and they visible on AWS.

cc @amine-khemissi @ivoryibuaka

Reply:
It sounds like these tags are internal implementation details (required for inner AWS-BigAnimal workings), and not so much for the customer to do something with them. 

If that is the case, we can mention that these tags exist, but I don't believe we need to add a list/table with the tag names here. Let me know what you think :) 

WDYT?

Reply:
Yes, I think I understand now. We don't need to list which tags come from BA by default, we just need to specify which prefixes and which keys are reserved, so they understand why they can't use them. I hope that makes sense?

In that case, I can add an expandable section with the list you provided above and a note saying these are the values they can't use.

Reply:
that is a good idea @gvasquezvargas

Reply:
Hi all, I have edited this section. Here is the preview so you can see how the expand is displayed. 
https://deploy-preview-5863--edb-docs-staging.netlify.app/docs/biganimal/latest/using_cluster/managing_cloud_account/csp_tagging/#considerations 

@navnathedb you mentioned you are going to put a link to docs from the UI? If yes, I have created a permanent URL redirect (PURL) that is more robust than the dynamic URL. The idea behind it is that, even if we make changes to the documentation and move this page around, there will always be a redirect in place to link to this content.  So for the the UI link we can use: https://www.enterprisedb.com/docs/purl/cloudservice/csp_tagging/#considerations 
Let me know if I can give you more information on this :). 
PD:  The redirect does not work right now, but it will work once we merge and publish this PR.

Reply:
We need to add doc link on the tag validation if error occurs. 
Detailed discussion happened on [Slack](https://edb-starlight.slack.com/archives/C02K4TSUCRJ/p1721834254729869).